web/src/search_suggestion.ts

```diff
@@ -597,20 +597,31 @@ function get_special_filter_suggestions(
 
     const last_string = Filter.unparse([last]).toLowerCase();
     suggestions = suggestions.filter((s) => {
+        let search_string = s.search_string;
         if (match_criteria(terms, s.incompatible_patterns)) {
             return false;
         }
         if (last_string === "") {
             return true;
         }
 
+        // If the suggestion is already negated and the search query is also negated,
+        // we should cancel out the negation while maintaining a suggestion match.
+        if (
+            (last.negated === true || is_search_operand_negated) &&
+            search_string.startsWith("--")
+        ) {
+            search_string = search_string.slice(1);
+            s.search_string = s.search_string.slice(2);
```

Comment:
Can you explain a bit more why you're keeping two copies of the search string here, and why you're slicing them on different indices? Maybe you can share an example of what would happen here.

Reply:
In the case of `-` with the suggestion of `-is:resolved`, we would get a suggestion string of `--is:resolved`. 
One copy of the string is to used to store the string that the typeahead should return. `is:resolved` 
The other copy of the string is used for pattern matching `-is:resolved`